### PR TITLE
memory statistics: add VmPeak

### PR DESCRIPTION
**Pull request by tjhei**

### CODE REVIEW

**Comment by tjhei** [comment]
I will need to update test output...

**Comment by tjhei** [comment]
well, except that this value is not going to be stable between runs. :-(

**Comment by tjhei** [comment]
@bangerth I don't know what to do about this. This is useful information, but it is not reliable between runs or different machines. The options are: disable vmpeak by default (annoying), not merge, or add a postprocess script for the unit test that deletes the vmpeak value. I think the last option is the easiest, but I haven't found time to do it...

**Comment by bangerth** [comment]
Let the same script do it that also deletes the timing information from the output files. It should be a one-line addition to some perl something.

**Comment by tjhei** [comment]
I hacked something together, but it is a bit more complicated than a "one-line addition".

**Comment by tjhei** [comment]
how about this?

File: source/postprocess/memory_statistics.cc [hunk 1]
```diff
@@ -44,17 +44,60 @@ namespace aspect
...
+
+    template <int dim>
+    void
+    MemoryStatistics<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Postprocess");
+      {
+        prm.enter_subsection("Memory statistics");
+        {
+          prm.declare_entry ("output virtual memory peak", "true",
```
Comment (gassmoeller) [inline source/postprocess/memory_statistics.cc]: Capitalize the input parameter

File: source/postprocess/memory_statistics.cc [hunk 2]
```diff
@@ -44,17 +44,60 @@ namespace aspect
...
+
+    template <int dim>
+    void
+    MemoryStatistics<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Postprocess");
+      {
+        prm.enter_subsection("Memory statistics");
+        {
+          output_vmpeak = prm.get_bool ("output virtual memory peak");
```
Comment (gassmoeller) [inline source/postprocess/memory_statistics.cc]: same here

File: source/postprocess/memory_statistics.cc (hunk 1 above)
Comment (bangerth) [inline source/postprocess/memory_statistics.cc]: Close -- we usually start parameter names with upper case letters.

Comment (bangerth) [reply source/postprocess/memory_statistics.cc]: Matter of fact, I think the proper name should be "Output peak virtual memory". Or maybe "Output peak virtual memory (VmPeak)".

**Comment by gassmoeller** [comment]
This must be the first simultaneous double review we had in a while, and we even had the same comment :smile:

**Comment by tjhei** [comment]
I decided that it is much more useful to do a ``max`` over all MPI ranks.

**Pull request merged**